pkg/protocol/sybilprotection/sybilprotectionv1/sybilprotection.go

```diff
@@ -46,54 +46,61 @@ type SybilProtection struct {
+			latestCommittedSlot := e.Storage.Settings().LatestCommitment().Slot()
+			latestCommittedEpoch := s.apiProvider.APIForSlot(latestCommittedSlot).TimeProvider().EpochFromSlot(latestCommittedSlot)
+			s.performanceTracker = performance.NewTracker(e.Storage.RewardsForEpoch, e.Storage.PoolStats(), e.Storage.Committee(), e.Storage.CommitteeCandidates, e.Storage.ValidatorPerformances, latestCommittedEpoch, e, s.errHandler, logger)
+			s.lastCommittedSlot = latestCommittedSlot
+
+			if s.optsInitialCommittee != nil {
+				if _, err := s.seatManager.RotateCommittee(0, s.optsInitialCommittee); err != nil {
+					panic(ierrors.Wrap(err, "error while registering initial committee for epoch 0"))
 				}
+			}
 
-				// When the engine is triggered initialized, snapshot has been read or database has been initialized properly,
-				// so the committee should be available in the performance manager.
-				e.InitializedEvent().OnTrigger(func() {
-					// Mark the committee for the last committed slot as active.
-					currentEpoch := e.CommittedAPI().TimeProvider().EpochFromSlot(e.Storage.Settings().LatestCommitment().Slot())
-					err := o.seatManager.InitializeCommittee(currentEpoch, e.Clock.Accepted().RelativeTime())
-					if err != nil {
-						panic(ierrors.Wrap(err, "error while initializing committee"))
-					}
-
-					o.InitializedEvent().Trigger()
-				})
+			// When the engine is triggered initialized, snapshot has been read or database has been initialized properly,
+			// so the committee should be available in the performance manager.
+			e.InitializedEvent().OnTrigger(func() {
+				// Mark the committee for the last committed slot as active.
+				currentEpoch := e.CommittedAPI().TimeProvider().EpochFromSlot(e.Storage.Settings().LatestCommitment().Slot())
+				err := s.seatManager.InitializeCommittee(currentEpoch, e.Clock.Accepted().RelativeTime())
+				if err != nil {
+					panic(ierrors.Wrap(err, "error while initializing committee"))
+				}
+
+				s.InitializedEvent().Trigger()
 			})
 
-			e.Events.SlotGadget.SlotFinalized.Hook(o.slotFinalized)
+			e.Events.SlotGadget.SlotFinalized.Hook(s.slotFinalized)
 
-			e.Events.SybilProtection.LinkTo(o.events)
+			e.Events.SybilProtection.LinkTo(s.events)
 
-			o.ShutdownEvent().OnTrigger(func() {
-				o.StoppedEvent().Trigger()
-			})
+			s.InitializedEvent().Trigger()
+		})
 
-			o.ConstructedEvent().Trigger()
+		return s
+	})
+}
+
+func New(subModule module.Module, engine *engine.Engine, opts ...options.Option[SybilProtection]) *SybilProtection {
+	return options.Apply(&SybilProtection{
+		Module:                  subModule,
+		events:                  sybilprotection.NewEvents(),
+		apiProvider:             engine,
+		optsSeatManagerProvider: topstakers.NewProvider(),
+	}, opts, func(s *SybilProtection) {
+		s.seatManager = s.optsSeatManagerProvider(engine)
+
+		s.ShutdownEvent().OnTrigger(func() {
+			s.StoppedEvent().Trigger()
 		})
+
+		s.ConstructedEvent().Trigger()
```

Comment:
Not constructed here because the Ledger, performance tracker and errHandler is set later and here it's still `nil`